app/include/zmk/behavior.h

```diff
@@ -42,6 +45,19 @@ struct zmk_behavior_binding_event {
  */
 const struct device *zmk_behavior_get_binding(const char *name);
 
+/**
+ * @brief Invoke a behavior given its binding and invoking event details.
+ *
+ * @param src_binding Behavior binding to invoke.
+ * @param event The binding event struct containing details of the event that invoked it.
+ * @param pressed Whether the binding is pressed or released.
+ *
+ * @retval 0 If successful.
+ * @retval Negative errno code if failure.
+ */
+int zmk_invoke_behavior_binding(const struct zmk_behavior_binding *src_binding,
```

Comment:
I would properly namespace this, e.g.:

```suggestion
int zmk_behavior_invoke_binding(const struct zmk_behavior_binding *src_binding,
```